homeassistant/components/august/__init__.py

```diff
@@ -75,138 +57,154 @@
+        result = await hass.async_add_executor_job(
+            august_gateway.authenticator.validate_verification_code, code
+        )
 
         if result == ValidationResult.INVALID_VERIFICATION_CODE:
-            configurator.notify_errors(
-                _CONFIGURING[DOMAIN], "Invalid verification code"
+            configurator.async_notify_errors(
+                _CONFIGURING[entry_id],
+                "Invalid verification code, please make sure you are using the latest code and try again.",
             )
         elif result == ValidationResult.VALIDATED:
-            setup_august(hass, config, api, authenticator, token_refresh_lock)
+            return await async_setup_august(hass, config_entry, august_gateway)
 
-    if DOMAIN not in _CONFIGURING:
-        authenticator.send_verification_code()
+        return False
+
+    if entry_id not in _CONFIGURING:
+        await hass.async_add_executor_job(
+            august_gateway.authenticator.send_verification_code
+        )
 
-    conf = config[DOMAIN]
-    username = conf.get(CONF_USERNAME)
-    login_method = conf.get(CONF_LOGIN_METHOD)
+    entry_data = config_entry.data
+    login_method = entry_data.get(CONF_LOGIN_METHOD)
+    username = entry_data.get(CONF_USERNAME)
 
-    _CONFIGURING[DOMAIN] = configurator.request_config(
-        NOTIFICATION_TITLE,
-        august_configuration_callback,
-        description=f"Please check your {login_method} ({username}) and enter the verification code below",
+    _CONFIGURING[entry_id] = configurator.async_request_config(
+        DEFAULT_NAME + " (" + username + ")",
+        async_august_configuration_validation_callback,
+        description="August must be re-verified. Please check your {} ({}) and enter the verification "
+        "code below".format(login_method, username),
         submit_caption="Verify",
         fields=[
-            {"id": "verification_code", "name": "Verification code", "type": "string"}
+            {"id": VERIFICATION_CODE_KEY, "name": "Verification code", "type": "string"}
         ],
     )
+    return
 
 
-def setup_august(hass, config, api, authenticator, token_refresh_lock):
+async def async_setup_august(hass, config_entry, august_gateway):
     """Set up the August component."""
 
-    authentication = None
     try:
-        authentication = authenticator.authenticate()
-    except RequestException as ex:
-        _LOGGER.error("Unable to connect to August service: %s", str(ex))
-
-        hass.components.persistent_notification.create(
-            "Error: {ex}<br />You will need to restart hass after fixing.",
-            title=NOTIFICATION_TITLE,
-            notification_id=NOTIFICATION_ID,
-        )
+        august_gateway.authenticate()
+    except RequireValidation:
+        await async_request_validation(hass, config_entry, august_gateway)
+        return False
+    except InvalidAuth:
+        _LOGGER.error("Password is no longer valid. Please set up August again")
```

Comment:
This is ok for now, in a future PR we should start a new config flow

Reply:
adding a comment